parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -79,10 +112,20 @@ public String toString() {
     private final int offset;
     private final int length;
 
+    @Deprecated
```

Comment:
This is a constructor in a private class, so I think you can safely delete the deprecated constructor.